src/core/stores/VNode.ts

```diff
@@ -483,6 +483,34 @@ export class VNode {
             return !node.hasChildren() && (!predicate || predicate(node));
         });
     }
+    /**
+     * Return all previous siblings. If a predicate is passed, return all
+     * previous siblings that satisfy its condition.
+     *
+     * @param [predicate]
+     * @param [__current] recursive collector
+     */
+    previousSiblings(predicate?: Predicate, __current: VNode[] = []): VNode[] {
+        if (!predicate || predicate(this)) {
+            __current.push(this);
+        }
+        const previous = this.previousSibling();
+        return previous ? previous.previousSiblings(predicate, __current) : __current;
```

Comment:
I don't really like using these obnoxious `__current` parameters. What do you think of this ?
```typescript
const prevSiblings: VNode[] = [];
let sibling = this.previousSibling();
while (sibling && (!predicate || predicate(sibling)) {
    prevSiblings.push(sibling);
    sibling = sibling.previousSibling();
}
return prevSiblings;
```

If we later want to implement this as an iterator, it would even be shorter:
```
typescript
let sibling = this.previousSibling();
while (sibling && (!predicate || predicate(sibling)) {
    yield sibling;
    sibling = sibling.previousSibling();
}
```